staging/src/k8s.io/code-generator/hack/lib/codegen.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+CODEGEN_PKG=${CODEGEN_PKG:-$(ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo "k8s.io/code-generator")}
+
+for gen in defaulter-gen conversion-gen client-gen lister-gen informer-gen deepcopy-gen conversion-gen; do
+  echo "Building ${gen}"
+  go install ${CODEGEN_PKG}/cmd/${gen}
+done
+
+function join { local IFS="$1"; shift; echo "$*"; }
+
+function generate-internal-groups() {
+  local BASE="$1" # the base packages are resolved against (empty or the vendor/ directory ${OUTPUT_PKG} is in)
+  local OUTPUT_PKG="$2" # the output package name (without a "foo/var/vendor/" prefix, if inside vendor/)
+  local INT_APIS_PKG="$3" # the internal types dir (e.g. k8s.io/kubernetes/pkg/apis)
+  local EXT_APIS_PKG="$4" # the external types dir (e.g. k8s.io/api; often equals the internal types dir)
+  local GROUPS_WITH_VERSIONS="$5" # groupA:v1,v2 groupB:v1 groupC:v2
+  shift 5
+
+  # enumerate group versions
+  local ALL_FQ_APIS=() # e.g. k8s.io/kubernetes/pkg/apis/apps k8s.io/api/apps/v1
+  local INT_FQ_APIS=() # e.g. k8s.io/kubernetes/pkg/apis/apps
+  local EXT_FQ_APIS=() # e.g. k8s.io/api/apps/v1
+  for GVs in ${GROUPS_WITH_VERSIONS}; do
+    IFS=: read G Vs <<<"${GVs}"
+
+    ALL_FQ_APIS+=(${INT_APIS_PKG}/${G})
+    INT_FQ_APIS+=(${INT_APIS_PKG}/${G})
+
+    # enumerate versions
+    for V in ${Vs//,/ }; do
+      ALL_FQ_APIS+=(${EXT_APIS_PKG}/${G}/${V})
+      EXT_FQ_APIS+=(${EXT_APIS_PKG}/${G}/${V})
+    done
+  done
+
+  echo "Generating deepcopy funcs"
+  ${GOPATH}/bin/deepcopy-gen -i $(join , "${ALL_FQ_APIS[@]}") -O zz_generated.deepcopy --bounding-dirs ${INT_APIS_PKG},${EXT_APIS_PKG} "$@"
```

Comment:
Where is boilerplate flag defined?

Reply:
Added a demonstration to sample-apiserver